src/python/pants/option/options_test.py

```diff
@@ -728,7 +713,9 @@ def test_arg_scoping() -> None:
 
 def test_unconsumed_args() -> None:
     with pytest.raises(UnknownFlagsError):
-        _parse(flags="--unregistered-option compile").verify_args()
+        # We set allow_unknown_options=True to ensure that it's the native
+        # parser that checks for unconsumed args, and not the legacy parser.
```

Comment:
Why does setting this arg have this effect? It seems a bit indirect?

Reply:
Ah, if we don't set it then it will fail earlier, in `for_scope()`, and never get to the explicit verification call.